CONTRIBUTING.md

```diff
@@ -3,21 +3,86 @@ Thanks for your interest in contributing to the project! Please follow these sim
-- Please don't add #Regions to the code
-- When adding new files, please prepend the standard license disclaimer (just copy and paste from another source file)
-- Do your best to follow these guidelines but don't be afraid to make mistakes when trying to apply them. We are all novices in the beginning.
+### `CefSharp` vs `CEF`
+
+`CefSharp` simply takes the building blocks provided by `Cef` and attempts to provide a usable '.Net' implementation.
+The upstream [`CEF` forum](http://magpcss.org/ceforum/) is a valuable resource, if your issues seems fairly low level, then please conduct some research before posting.
+
+It maybe helpful to run the `cefclient` application and compare output with `CefSharp`. The `WinForms` and `WPF` versions use two different rending modes, `WPF` uses Offscreen Rendering (`OSR`).
+
+- Download **Test App** from http://cefbuilds.com
+- To compare with WPF run `cefclient --multi-threaded-message-loop --off-screen-rendering-enabled`
+- To compare with WinForms `cefclient --multi-threaded-message-loop`
+
+To determine which version of `CEF` your build is running, open `chrome://version` and you'll see a number similar to `3.2062.1898`, in this case `2062` represents the branch which corresponds to http://cefbuilds.com/#branch_2062
+
+### What should I include when creating an `Issue`?
+
+A bug is a _demonstrable problem_ that is caused by the code in the repository. Ideally each issue is a useful resource for references purposes (Don't take offence if someone edits your description).
+
+1. Good bug reports are extremely helpful. The more information you provide, the more likely your issue will be resolved.
+2. Good bug reports shouldn't leave others needing to chase you up for more information. Be sure to include the
+details of your environment.
+3. 'Ask Don't Tell' : Ask how to achieve something, **don't say it's broken** just because you haven't got it working yet!
+4. **Isolate the problem** &mdash; ideally create a reproducible example.
+5. **Include a screencast if relevant** - Is your issue about a design or front end feature or bug? The most helpful thing in the world is if we can *see* what you're talking about. Use [LICEcap](http://www.cockos.com/licecap/) to quickly and easily record a short screencast (24fps) and save it as an animated gif! Embed it directly into your GitHub issue.
+6. **Include as much info as possible!** Use the **Bug Report template** below or [click this link](https://github.com/CefSharp/CefSharp/issues/new?title=Bug%3A&body=%23%23%23%20Issue%20Summary%0A%0A%23%23%23%20Steps%20to%20Reproduce%0A%0A1.%20This%20is%20the%20first%20step%0A%0AThis%20is%20a%20bug%20because...%0A%0A%23%23%23%20Technical%20details%0A%0A*%20Ghost%20Version%3A%20master%20-%20latest%20commit%3A%20%20INSERT%20COMMIT%20REF%0A*%20Client%20OS%3A%20%0A*%20Server%20OS%3A%20%0A*%20Node%20Version%3A%20%0A*%20Browser%3A%20%0A*%20Database%3A) to start creating a bug report with the template automatically.
```

Comment:
You got in before I had a chance to fix that :smile: 

See https://github.com/amaitland/CefSharp/commit/9022ec423cbde386e91f21bdf5c2c7d3d9628d46